### PR TITLE
chore(all): Remove the box_syntax feature gate.

### PR DESCRIPTION
**Pull request by reem**

### CODE REVIEW

File: src/client/request.rs [hunk 1]
```diff
@@ -56,7 +56,7 @@ impl Request<Fresh> {
         let (host, port) = try!(get_host_and_port(&url));
 
         let stream = try!(connector.connect(&*host, port, &*url.scheme));
-        let stream = ThroughWriter(BufWriter::new(box stream as Box<NetworkStream + Send>));
+        let stream = ThroughWriter(BufWriter::new(Box::new(stream) as Box<NetworkStream + Send>));
```
Comment (seanmonstar) [inline src/client/request.rs]: This works now? I kept the `box` syntax cause there didn't used to be a way to coerce to a Boxed TraitObject without it...

**Comment by seanmonstar** [comment]
I can't merge this? Did you start from an older master?

File: src/client/request.rs (hunk 1 above)
Comment (reem) [inline src/client/request.rs]: IIRC it's worked ever since `Box::new` existed, but the silent coercions with `Box::new` aren't as ambitious as with `box`.

**Comment by reem** [comment]
Possibly, let me rebase.

**Comment by seanmonstar** [comment]
I've included this in #403

**Pull request closed without merging**